src/styles/calendar.scss

```diff
@@ -0,0 +1,44 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: 1px;
+  width: 720px;
+  padding: 10px;
+
+  &__day {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    border: 1px solid black;
+    height: $height-cell;
+    width: $width-cell;
+    background-color: #eee;
+    font-family: Arial, Helvetica, sans-serif;
+    font-size: 30px;
+    transition: 0.5s;
+
+    &:hover {
+      cursor: pointer;
+      background-color: pink;
+      transform: translateY(-20px);
+    }
+  }
+
+  @for $num from 1 through 31 {
+    &__day:nth-child(#{$num})::before {
+      content: "#{$num}";
+    }
+  }
+
+  @each $day in $week {
+    .start-day--#{day}:first-child {
+      margin-left: #{margin};
+    }
+  }
+
+  @for $i from 28 through 31 {
+    &--month-length-#{$i}:nth-child(n + #{$i + 1}) {
+      display: block;
```

Comment:
if modifier has value 29, then this loop should hide 30, 31 cells, what are you trying to achieve using display block?